src/connection.js

```diff
@@ -1274,6 +1306,95 @@ class Connection extends EventEmitter {
     this.transitionTo(this.STATE.SENT_LOGIN7_WITH_STANDARD_LOGIN);
   }
 
+  getRequestOptions() {
+    var tokenHost = '169.254.169.254';
+    var tokenEndpoint = '/metadata/identity/oauth2/token';
+    var clinetIdKeyName = '&client_id=';
+    const apiVersionKeyName = '&api-version=';
+    const clinetId = this.config.authentication.options.MSIclientID;
+    const msiEndpoint = process.env.MSI_ENDPOINT;
+    const msiSecret = process.env.MSI_SECRET;
+    var tokenPath = '';
+    var tokenPort = '';
+    var tokenHeaders = { 'Metadata': true };
+    // Api version supported on Azure VM
+    var apiVersion = '2018-02-01';
+
+    // If MSI_ENDPOINT and MSI_SECRET are defined, used these instead of the hard-coded host
+    if (undefined !== msiEndpoint && undefined !== msiSecret) {
+      // Parse the environment variable as a url and set corresponding request components
+      var appUrl = url.parse(msiEndpoint, true);
+      tokenHost = appUrl.hostname;
+      tokenPath = appUrl.pathname;
+      tokenPort = appUrl.port;
+      tokenHeaders = {
+        'Secret': msiSecret
+      };
+      // Api version supported on Azure App
+      apiVersion = '2017-09-01';
+      clinetIdKeyName = '&clientid=';
```

Comment:
I think this might be a typo? You probably meant to call this `clientIdKeyName` 😬